revolv/project/models.py

```diff
@@ -481,14 +490,56 @@ def partial_repayment_as_js(self):
         return unicode(self.partial_repayment)
 
     @property
+    def total_days(self):
+        """
+        :return the total length of the campaign of this project,
+        or None if the project hasn't started yet.
+
+        Note: if a project's campaign starts and ends on the same day, it is
+        defined to be one day long, not zero days long.
+        """
+        if self.start_date is None:
+            return None
+        return max((self.end_date - self.start_date).days + 1, 0)
+
+    @property
     def days_until_end(self):
-        return (datetime.date.today() - self.end_date).days
+        """
+        :return: the difference between today and the end date of this project.
+        May be negative.
+        """
+        return (self.end_date - datetime.date.today()).days
+
+    @property
+    def days_so_far(self):
+        """
+        :return: the integer number of days that have passed since
+        this project's campaign began, or None if it has not started
+        yet.
+        """
+        if self.start_date is None:
+            return None
+        difference = (datetime.date.today() - self.start_date).days
+        if difference < 0:
+            return 0
+        if difference > self.total_days:
+            return self.total_days
+        return difference
 
     @property
     def days_left(self):
+        """
+        :return: the integer number of days until the end of this project,
+        or 0 if the project's campaign has finished.
+        """
         return max(self.days_until_end, 0)
 
     def formatted_days_left(self):
+        """
```

Comment:
if this is being used in a template, wouldn't it be better to format this in the template?